dotCMS/src/main/java/com/dotmarketing/portlets/rules/parameter/type/DataType.java

```diff
@@ -77,5 +89,57 @@ public void checkValid(String value){
     public Map<String, TypeConstraint> getConstraints() {
         return ImmutableMap.copyOf(this.restrictions);
     }
-}
- 
+
+    /**
+     * Utility type used to correctly read immutable object from JSON representation.
+     *
+     * @deprecated Do not use this type directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    @Deprecated
+    @JsonDeserialize
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+    @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+    static final class Json {
+
+        @javax.annotation.Nullable
+        String id;
+        @javax.annotation.Nullable
+        String errorMessageKey;
+        @javax.annotation.Nullable
+        Map<String, TypeConstraint> restrictions;
+
+        @JsonProperty("id")
+        public void setKey(@Nullable String id) {
+            this.id = id;
+        }
+
+        @JsonProperty("errorMessageKey")
+        public void setDataType(@Nullable String errorMessageKey) {
+            this.errorMessageKey = errorMessageKey;
+        }
+
+        @JsonProperty("constraints")
+        public void setRestrictions(@Nullable Map<String, TypeConstraint> restrictions) {
+            this.restrictions = restrictions;
+        }
+
+    }
+
+    /**
+     * @param json A JSON-bindable data structure
+     * @deprecated Do not use this method directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    @Deprecated
+    @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
+    static DataType fromJson(DataType.Json json) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'json' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)